inventory_management_system_api/services/utils.py

```diff
@@ -23,3 +33,152 @@ def generate_code(name: str, entity_type: str) -> str:
     logger.info("Generating code for the %s based on its name", entity_type)
     name = name.lower().strip()
     return re.sub(r"\s+", "-", name)
+
+
+def process_catalogue_item_properties(
+    defined_properties: List[CatalogueItemProperty],
+    supplied_properties: List[PropertyPostRequestSchema],
+) -> List[Dict]:
+    """
+    Process and validate supplied catalogue item properties based on defined catalogue item properties. It checks
+    for missing mandatory catalogue item properties unless otherwise instructed, filters the matching catalogue item
+    properties, adds the property units, and finally validates the property values.
+
+    The `supplied_properties_dict` dictionary may get modified as part of the processing and validation.
+
+    :param defined_properties: The list of defined catalogue item property objects.
+    :param supplied_properties: The list of supplied catalogue item property objects.
+    :return: A list of processed and validated supplied catalogue item properties.
+    """
+    # Convert properties to dictionaries for easier lookups
+    defined_properties_dict = _create_catalogue_item_properties_dict(defined_properties)
+    supplied_properties_dict = _create_catalogue_item_properties_dict(supplied_properties)
+
+    # Some mandatory catalogue item properties may not have been supplied
+    _check_missing_mandatory_catalogue_item_properties(defined_properties_dict, supplied_properties_dict)
+    # Catalogue item properties that have not been defined may have been supplied
+    supplied_properties_dict = _filter_matching_catalogue_item_properties(
+        defined_properties_dict, supplied_properties_dict
+    )
```

Comment:
Is there a reason you chose this over a 422 or something if more properties are supplied? The only reason I can see a problem is if an optional property has a wrong spelling or something it will just ignore it rather than complain.

Reply:
It mimics the behaviour of Pydantic. If you provide extra fields to Pydantic that do not match any of the defined fields, it will just ignore them.